core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java

```diff
@@ -240,11 +242,30 @@ public String toString() {
       return getClass().getSimpleName() + "[Outgoing marshaller: " + req_marshaller + "; incoming marshaller: " + rsp_marshaller + "]";
    }
 
-   private void reply(org.jgroups.blocks.Response response, Object retVal, ReplicableCommand command) {
+   private void reply(org.jgroups.blocks.Response response, Object retVal, ReplicableCommand command,
+         Message req) {
       if (response != null) {
          if (trace) log.tracef("About to send back response %s for command %s", retVal, command);
+         Buffer rsp_buf;
+         boolean is_exception = false;
+         try {
+            rsp_buf = rsp_marshaller.objectToBuffer(retVal);
+         } catch (Throwable t) {
+            try {  // this call should succeed (all exceptions are serializable)
+               rsp_buf = rsp_marshaller.objectToBuffer(t);
+               is_exception = true;
+            } catch (Throwable tt) {
+               log.errorMarshallingObject(tt, retVal);
+               return;
+            }
+         }
+
+         // Always set the NO_FC flag
+         short flags = (short) (req.getFlags() | NO_FC.value() & ~Message.Flag.RSVP.value() & ~Message.Flag.SCOPED.value());
```

Comment:
I would remove the ```INTERNAL``` flag and set ```NO_TOTAL_ORDER``` flag (I don't see a case in ISPN where we need to send the reply in TO :) )